src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -209,6 +218,17 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     isSolutionAvailable,
                     deferredProjectsData.PackageSpecs,
                     token);
+
+                // To limit risk, we only publish the event when there is a cross-platform PackageReference project
+                // in the solution. Extending this behavior to all solutions is tracked here:
+                // https://github.com/NuGet/Home/issues/4478
+#if VS15
```

Comment:
better option would be to have !VS14 so that we dont need to change anything for next VS versions